spec/persister/finders_spec.rb

```diff
@@ -19,7 +19,7 @@
   it "raises an exception when relation object is needed, but something else is provided" do
     expected_error = "Wrong index for key :vm_or_template, the value must be of type Nil or InventoryObject or InventoryObjectLazy, got: not_allowed_string"
     expect do
-      persister.hardwares.lazy_find(:vm_or_template => "not_allowed_string")
+      persister.hardwares.lazy_find({:vm_or_template => "not_allowed_string"})
```

Comment:
This is going to require _a lot_ of changes in MIQ if we have to change the callers...

Reply:
True, but this is only because `lazy_find` accepts a hash as the first argument and optional keyword arguments for the rest.   Maybe I'm naive but I believe we generally used options hashes as the last argument and didn't really adopt keyword arguments generally. 

The changes in [core here](https://github.com/ManageIQ/manageiq/pull/21531/files) are pretty small but it's possible we're not really exercising much of the code and we could find others like this in other repositories.

Reply:
I think what's confusing is that the first parameter is named manager_uid, when in fact it accepts a hash.

Reply:
`manager_ref` would be more accurate, and if manager_ref is only a single column you can just pass a string to lazy_find for simplicity

Reply:
I wonder if it's better to make that a keyword arg also, so something like:

```ruby
def lazy_find(manager_ref:, ref: primary_index_ref, key: nil, default: nil, transform_nested_lazy_finds: false) 
##############
      persister.hardwares.lazy_find(manager_ref: {:vm_or_template => "not_allowed_string"})
```

Reply:
Okay just checked and the number of times we pass a hash in to lazy_find is relatively small (22 out of 335 total callers)

Reply:
> I wonder if it's better to make that a keyword arg also, so something like:

Maybe?  Might just be lazy but `vms.lazy_find("vm-123")` is pretty nice compared with having to pass `vms.lazy_find(manager_ref: "vm-123")`

Reply:
If we're doing a major version bump anyway...these methods have _way_ too many options haha
Almost all of these could be handled automatically, the only one that can't is `:default` but I'm not sure if that's used currently

Reply:
yeah agreed...the signatures here have always been pretty nasty.

Reply:
Is the first parameter to `lazy_find` optional?
If it is non-optional, then I think ruby will be smart enough to know it is the first parameter and not part of the kwargs.

Other than that, think this is the new style of ruby and the path we need to take

Reply:
> Is the first parameter to lazy_find optional?

No, it's a required positional argument.

```ruby
def lazy_find(manager_ref:, ref: primary_index_ref, key: nil, default: nil, transform_nested_lazy_finds: false) 
```